src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -525,6 +525,16 @@ def get_access_token_for_resource(self, username, tenant, resource):
             username, tenant, resource)
         return access_token
 
+    def _get_ssh_certificate_for_resource(self, tenant, modulus, exponent):
```

Comment:
>ssh_certificate [](start = 13, length = 15)

please just implement a method like get_msal_for_resource and put the SSH related logic in vm module